evm/src/main/java/org/hyperledger/besu/evm/operation/PayOperation.java

```diff
@@ -0,0 +1,102 @@
+package org.hyperledger.besu.evm.operation;
+
+import static org.hyperledger.besu.evm.internal.Words.clampedAdd;
+import static org.hyperledger.besu.evm.operation.AbstractExtCallOperation.EOF1_EXCEPTION_STACK_ITEM;
+import static org.hyperledger.besu.evm.operation.AbstractExtCallOperation.EOF1_SUCCESS_STACK_ITEM;
+
+import java.util.Objects;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.Code;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.account.Account;
+import org.hyperledger.besu.evm.account.MutableAccount;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.internal.Words;
+
+public class PayOperation extends AbstractOperation {
+  static final int OPCODE = 0xfc;
+  /**
+   * Instantiates a new Abstract operation.
+   *
+   * @param gasCalculator the gas calculator
+   */
+  public PayOperation(final GasCalculator gasCalculator) {
+    super(OPCODE, "PAY", 2, 1, gasCalculator);
+  }
+
+  @Override
+  public OperationResult execute(final MessageFrame frame, final EVM evm) {
+    Code callingCode = frame.getCode();
+    if (callingCode.getEofVersion() == 0) {
+      return InvalidOperation.INVALID_RESULT;
+    }
+
+    final Bytes toAddressBytes = frame.getStackItem(1);
+    if (toAddressBytes.size() > 20
+      && toAddressBytes.numberOfLeadingZeroBytes() < 12) {
```

Comment:
Does the stack trim zeros by default? If not, what if I push 21 bytes to the stack with a leading 0 and an address of 20 nonzero bytes? This throws, but should be valid.

Reply:
`PUSH0 PUSH21 00<20 nonzero bytes> PAY` maybe?